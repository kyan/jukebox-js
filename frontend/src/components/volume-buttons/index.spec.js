import React from 'react'
import { Provider } from 'react-redux'
import configureMockStore from 'redux-mock-store'
import { mount } from 'enzyme'
import VolumeButtons from './index'

describe('VolumeButtons', () => {
  const volMock = jest.fn()
  const mockStore = configureMockStore()
  const buildWrapper = (store, props) => (
    mount(
      <Provider store={store}>
        <VolumeButtons
          {...props}
          onVolumeChange={volMock}
        />
      </Provider>
    ).find('VolumeButtons')
  )

  beforeEach(() => {
    jest.clearAllMocks()
  })

  describe('render', () => {
    it('render', () => {
      const store = mockStore({ jukebox: { volume: 0 } })
      const wrapper = buildWrapper(store, { disabled: false })
      expect(wrapper).toMatchSnapshot()
    })

    it('handles a volume down click', () => {
      const store = mockStore({ jukebox: { volume: 30 } })
      const wrapper = buildWrapper(store, { disabled: false })
      wrapper.find('VolumeDownButton').simulate('click')
      expect(volMock).toHaveBeenCalledWith(25)
    })

    it('handles a volume down click when min is reached', () => {
      const store = mockStore({ jukebox: { volume: 0 } })
      const wrapper = buildWrapper(store, { disabled: false })
      wrapper.find('VolumeDownButton').simulate('click')
      expect(volMock).not.toHaveBeenCalled()
    })

    it('handles a volume up click', () => {
      const store = mockStore({ jukebox: { volume: 30 } })
      const wrapper = buildWrapper(store, { disabled: false })
      wrapper.find('VolumeUpButton').simulate('click')
      expect(volMock).toHaveBeenCalledWith(35)
    })

    it('handles a volume up click when max is reached', () => {
      const store = mockStore({ jukebox: { volume: 100 } })
      const wrapper = buildWrapper(store, { disabled: false })
      wrapper.find('VolumeUpButton').simulate('click')
      expect(volMock).not.toHaveBeenCalled()
    })

    it('handles a button being disabled', () => {
      const store = mockStore({ jukebox: { volume: 32 } })
      const wrapper = buildWrapper(store, { disabled: true })
      wrapper.find('VolumeUpButton').simulate('click')
      expect(volMock).not.toHaveBeenCalled()
    })
  })
})
