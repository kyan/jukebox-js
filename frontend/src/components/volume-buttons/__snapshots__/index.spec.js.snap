// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VolumeButtons render render 1`] = `
<VolumeButtons
  disabled={false}
  onVolumeChange={[MockFunction]}
>
  <div
    className="c-volumeControls"
  >
    <VolumeDownButton
      disabled={false}
      onChange={[MockFunction]}
      volume={0}
    >
      <button
        className="c-button"
        disabled={false}
      >
        <Icon
          as="i"
          name="volume down"
        >
          <i
            aria-hidden="true"
            className="volume down icon"
            onClick={[Function]}
          />
        </Icon>
      </button>
    </VolumeDownButton>
    <VolumeUpButton
      disabled={false}
      onChange={[MockFunction]}
      volume={0}
    >
      <button
        className="c-button"
        disabled={false}
        onClick={[Function]}
      >
        <Icon
          as="i"
          name="volume up"
        >
          <i
            aria-hidden="true"
            className="volume up icon"
            onClick={[Function]}
          />
        </Icon>
      </button>
    </VolumeUpButton>
    <div
      className="c-volumeControls__indicator"
    >
      0
    </div>
  </div>
</VolumeButtons>
`;
